pkcs11/src/new/types/mechanism/mod.rs

```diff
@@ -1,11 +1,35 @@
 mod rsa;
 
 use crate::new::Error;
+use log::error;
 use pkcs11_sys::*;
 use std::convert::TryFrom;
+use std::ops::Deref;
 use std::ptr::null_mut;
 
-pub enum Mechanism {
+const RSA_PKCS_KEY_PAIR_GEN: MechanismType = MechanismType {
+    val: CKM_RSA_PKCS_KEY_PAIR_GEN,
+};
+const RSA_PKCS: MechanismType = MechanismType { val: CKM_RSA_PKCS };
+
+#[derive(Debug, PartialEq, Eq)]
+// transparent so that a vector of MechanismType should have the same layout than a vector of
+// CK_MECHANISM_TYPE.
+#[repr(transparent)]
+pub struct MechanismType {
```

Comment:
`MechanismType` can not be an `enum` because it must contain a `CK_MECHANISM_TYPE` value to be passed in an attribute... To overcome this, it does not have a constructor and contain a public list of `const` values to be used by users.